mitodl/ocw-hugo-themes
# feat: featured courses

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/526

#### What's this PR do?
- Creates a new partial to get featured courses which fetches the featured courses from a `course list` whose title is `featured-courses-homepage`. (Note: It was decided to keep the list hard-coded for now)
- Creates a new partial to render/display featured courses and uses that partial in home page.
- Courses from these `featured-courses` course list are fetched from `staticApiBaseUrl/course/data.json`
- Then these courses are rendered.
- If no course or list is found then whole section is not rendered. 

#### How should this be manually tested?
- Go to either studio RC or on local setup of studio
- Create a course list with title `featured-courses` and add some courses for `ocw-www`
- Publish/Clone `ocw-www`
- Verify that now featured courses are being rendered on www home page
- Verify that all information displayed on featured course card is accurate.
- Verify that all links are working correctly.
- Verify that images are rendering correctly.
- Verify any other thing that need to be looked upon

#### Screenshots

Please ignore the missing images as I haven't set my local setup to run both, www and courses on the same server, hence the images having path: `/courses/course-id/image.jpg` are throwing 404. (Note: this is only on my local setup, will work fine on RC/Prod)

![www homepage desktop](https://user-images.githubusercontent.com/93309234/160881572-a11f26fa-58a4-43e6-b82f-d953a12a3ebd.png)

![www homepage mobile](https://user-images.githubusercontent.com/93309234/160881592-3021fcaa-b0fc-4477-8439-bb7d9418861e.png)
